swig/lightgbmlib.i

```diff
@@ -127,7 +135,13 @@
     jenv->ReleasePrimitiveArrayCritical(values, values0, JNI_ABORT);
     jenv->ReleasePrimitiveArrayCritical(indices, indices0, JNI_ABORT);
 
-    return ret;
+    if (ret != 0) {
+      return nullptr;
+    }
+    jdoubleArray new_array = jenv->NewDoubleArray(*out_len);
```

Comment:
does this allocate memory underneath? I'm confused by doc :)

Reply:
yes, it should return a new double array

Reply:
I am worried about allocating both a new array on native side and also on java side, maybe I should stick to reusing the same native array (and passing it in as arg?) but returning a new java array